### ISSUE TITLE
Offset and limit ignored in subquery

### ISSUE THREAD

**Issue by tino415** [yiisoft/yii2#8266]
When I use subquery, offset and limit arent in generated sql (postgresql):

Original query:

```sql
SELECT * FROM tag WHERE id IN (
    SELECT tag_id FROM (
        SELECT * FROM view LIMIT :limit OFFSET :offset
    ) AS cluster
    GROUP BY tag_id
    ORDER BY COUNT(*)
 )
LIMIT :limit_per
```

PHP code:
```php
$querySlice = (new Query)->select('*')
    ->from('view')
    ->limit($cluster_size)
    ->offset($i);

$topOfSlice = (new Query)->select('tag_id')
    ->from(['cluster' => $querySlice])
    ->groupBy('tag_id')
    ->orderBy(new Expression('COUNT(*)'));

$tags = Tag::find()
    ->where(['id' => $topOfSlice])
    ->limit($per_cluster_top)
    ->all();
```

Generated query from debugger:

```sql
SELECT * FROM "tag" WHERE "id" IN (
    SELECT "tag_id" FROM (
        SELECT * FROM "view"
    ) "cluster" 
    GROUP BY "tag_id" 
    ORDER BY COUNT(*)
) 
LIMIT 10
```

**Comment by tino415** [comment]
Limit not working if I set it as string because hasLimit and hasOffset function in QueryBuilder

**Comment by nineinchnick** [comment]
So it's related to or duplicate of #7023?

**Comment by nineinchnick** [comment]
But it should work with strings, there's an explicit condition for it.

**Comment by tino415** [comment]
Yes i found it now

**Comment by tino415** [comment]
Shouldnt by better if it accept anithing convertable to unsigned integere ?

**Comment by cebe** [comment]
closed as duplicate of #7023
